app/models/job_proxy_dispatcher.rb

```diff
@@ -281,7 +281,7 @@ def get_eligible_proxies_for_job(job)
     end
 
     if @vm.storage.nil?
-      unless ['Amazon', 'OpenStack'].include?(@vm.vendor)
+      unless %w(Amazon OpenStack Microsoft).include?(@vm.vendor)
```

Comment:
@roliveri would be good (in a future PR) to ask the `@vm` object a question instead of having the knowledge here of which vendors are supported.